doc/en/build_rules/idl.md

```diff
@@ -39,6 +39,26 @@ proto_library(
 
 The `cpp` target code is always generated.
 
+If you want to generate some cpp code with other suffix names, you can use `cpp_outs` attributes.
+The default value is `['.pb']`. For example, if you want to use grpc plugin to generate `rpc_meta_info.pb.cc/h` and `rpc_meta_info.grpc.pb.cc/h`, set `cpp_outs = [".pb', '.grpc.pb']`
+
+Besides, use `plugin_opts` to pass some options to plugins, like `plugin_opts={"plugin_name":["option1", "option2.."]},`.
```

Comment:
These two attributes are actually belonged to the plugin. Can they be defined as plugin-level attributes ? By this way, they don't need to be written repeatedly for each proto_library call.

Reply:
``cpp_outs and plugin_opts would be differnent in differnent targets. The plugin will fragment the original file into multiple pieces based on the size of the proto generated, which the user will specify in different targets, the default is divided into 3 pieces.

Here is the code we will add in BLADE_ROOT:
```
if get_env("USE_ADTABLE_PLUGIN_FAST_MODE"):
    def _register_proto_library_wrapper():
        from blade import proto_library_target
        from blade import build_rules
        def proto_library(*args, **kwargs):
            default_opts = ["fast_mode", "sharding_size=3"]
            default_cpp_outs = [".pb", ".core.pb", ".controller.pb", ".axu.pb", ".core.1.pb", ".core.2.pb", ".core.3.pb"]
            default_applied = False
            if "plugins" in kwargs and "adtable_codegen" in kwargs["plugins"]:
                if "plugin_opts" not in kwargs:
                    kwargs["plugin_opts"] = {"adtable_codegen":default_opts}
                    default_applied = True
                elif "adtable_codegen" not in kwargs["plugin_opts"]:
                    kwargs["plugin_opts"]["adtable_codegen"] = default_opts
                    default_applied = True
            if default_applied:
                if "cpp_outs" not in kwargs:
                    kwargs["cpp_outs"] = []
                kwargs["cpp_outs"].extend(default_cpp_outs)
                kwargs["cpp_outs"] = list(set(kwargs["cpp_outs"])) # unique
            return proto_library_target.proto_library(*args, **kwargs)
        build_rules.register_function(proto_library)
    _register_proto_library_wrapper()
```